runtime/src/utils/buffer/write.rs

```diff
@@ -2,104 +2,84 @@ use crate::{Blob, Error, RwLock};
 use commonware_utils::StableBuf;
 use std::sync::Arc;
 
-/// The internal state of a [Write] buffer.
-struct Inner<B: Blob> {
-    /// The underlying blob to write to.
-    blob: B,
-    /// The buffer storing data to be written to the blob.
-    ///
-    /// The buffer always represents data at the "tip" of the logical blob,
-    /// starting at `position` and extending for `buffer.len()` bytes.
-    buffer: Vec<u8>,
+/// The state of any buffered data yet to be written to a [Blob].
+///
+/// The buffer always represents data at the "tip" of the logical blob, starting at `position` and
+/// extending for `data.len()` bytes.
+struct Buffer {
+    /// The data to be written to the blob.
+    data: Vec<u8>,
     /// The offset in the blob where the buffered data starts.
     ///
-    /// This represents the logical position in the blob where `buffer[0]` would be written.
-    /// The buffer is maintained at the "tip" to support efficient size calculation and appends.
+    /// This represents the logical position in the blob where `data[0]` would be written. The
+    /// buffer is maintained at the "tip" to support efficient size calculation and appends.
     position: u64,
     /// The maximum size of the buffer.
     capacity: usize,
 }
 
-impl<B: Blob> Inner<B> {
+impl Buffer {
     /// Returns the current logical size of the blob including any buffered data.
     fn size(&self) -> u64 {
-        self.position + self.buffer.len() as u64
+        self.position + self.data.len() as u64
     }
 
-    /// Appends bytes to the internal buffer, maintaining the "buffer at tip" invariant.
+    /// Appends `buf` to the internal `data` buffer, maintaining the "buffer at tip" invariant.
     ///
-    /// If the buffer capacity would be exceeded, it is flushed first. If the data
-    /// is larger than the buffer capacity, it is written directly to the blob.
+    /// If the buffer capacity would be exceeded, it is flushed first. If the data is larger than
+    /// the buffer capacity, it is written directly to the blob.
     ///
-    /// Returns an error if the write to the underlying [Blob] fails (may be due to a `flush` of data not
-    /// related to the data being written).
-    async fn write<S: Into<StableBuf>>(&mut self, buf: S) -> Result<(), Error> {
+    /// Returns an error if the write to `blob` fails (may be due to a `flush` of data not related
+    /// to the data being written).
+    async fn write(
+        &mut self,
+        blob: &impl Blob,
```

Comment:
I felt it was the right abstraction to give ownership of the `Blob` to the `Buffer` to avoid this (error-prone) pattern where the buffer for a blob can be applied to an arbitrary blob.

I'm not sure if it makes sense to keep it as it was (and just be able to `take()` from here) but figured I'd share the rationale.